src/views/NominatePeers/NominatePeers.js

```diff
@@ -1,73 +1,48 @@
+import { LoadingSpinner } from '../../components/LoadingSpinner';
 import NominatePeersModal from '../../components/Modal/NominatePeersModal';
 
-
-const columns = [
-  {
-    name: <h4>Name</h4>,
-    selector: row => row.fullname,
-    width: '300px',
-    sortable: true,
-  },
-  {
-    name: <h4>Division</h4>,
-    selector: row => row.division,
-    sortable: true,
-  },
-  {
-    name: <h4>Level</h4>,
-    selector: row => row.level,
-    sortable: true,
-  },
-  {
-    name: <h4>Status</h4>,
-    selector: 'status',
-    cell: row => (
-      <span style={{ color: row.status === 'Nominated' ? 'navy' : 'darkred' }}>
-        {row.status}
-      </span>
-    ),
-    sortable: true,
-  },
-  {
-    name: <h4>Actions</h4>,
-    cell: row => row.actions,
-  },
-];
-
 function NominatePeers() {
-  const history = useHistory();
   const [listUser, setListUser] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
-
   const authUser = useSelector(state => state.auth.user);
 
   const handleNominatePeer = async (userId) => {
     try {
-      // const selectedUser = listUser.find(user => user.id === userId);
-      // if (!selectedUser) throw new Error("User not found");
+      console.log("Nominate User ID:", userId);
+      console.log("List of Users:", listUser);
+
+      listUser.forEach(user => {
+        console.log(`User ID: ${user.id}, Status: ${user.status}, Level: ${user.level}`);
+      });
 
-      // console.log("Nominate User ID:", userId);
-      // console.log("Subordinates:", subordinates);
-      // const selectedUser = subordinates.find(user => user.id === userId);
-      // if (!selectedUser) throw new Error("User not found");
+      const selectedUser = listUser.find(user => user.id === userId);
+      if (!selectedUser) throw new Error("User not found");
 
-      // const nominatedStaff = subordinates.filter(user => user.status === 'nominated' && user.level === 'staff').length;
-      // if (selectedUser.level === 'staff' && nominatedStaff >= 3) {
-      //   throw new Error("You can't nominate more than 3 staff");
-      // }
+      const nominatedStaff = listUser.filter(user => user.status === 'Nominated' && user.level === 'Staf').length;
+      const nominatedKadiv = listUser.filter(user => user.status === 'Nominated' && user.level === 'Kepala Divisi').length;
+      const nominatedKanit = listUser.filter(user => user.status === 'Nominated' && user.level === 'Kepala Unit').length;
+
+      console.log('cek ini', nominatedStaff)
+
+      if (selectedUser.level === 'Staf' && nominatedStaff > 3) {
+        throw new Error("You can't nominate more than 3 staff members");
+      }
+      else if (selectedUser.level === 'Kepala Divisi' && nominatedKadiv > 3 ) {
+        throw new Error("You can't nominate more than 3 Kepala Divisi");
+      }
+      else if (selectedUser.level === 'Kepala Unit' && nominatedKanit > 2 ) {
+        throw new Error("You can't nominate more than 2 Kepala Unit");
+      }
```

Comment:
:wrench: Remove console log and create a new function for validation (refactor)

The handleNominatePeer should then just call your function

Reply:
handleNominatePeer {
   ...
   validateNominateUser()
   // your rest of code
}